src/test/java/nl/tudelft/contextproject/camera/LiveCameraConnectionTest.java

```diff
@@ -77,94 +40,209 @@ public void testSetUpConnection() {
+            assertWithinMaxOffset(before[0] - 420, after[0]);
+            assertWithinMaxOffset(before[1] - 420, after[1]);
         }
     }
 
     @Test
-    public void testRelTilt() {
+    public void testRelPan() throws InterruptedException {
         if (doTests) {
-            fail("Not yet implemented");
+            int[] before = connection.getCurrentPanTilt();
+            assertTrue(connection.relPan(420));
+            Thread.sleep(2000);
+            int[] after = connection.getCurrentPanTilt();
+            assertWithinMaxOffset(before[0] + 420, after[0]);
+            assertWithinMaxOffset(before[1], after[1]);
         }
     }
 
     @Test
-    public void testRelZoom() {
+    public void testRelTilt() throws InterruptedException {
         if (doTests) {
-            fail("Not yet implemented");
+            int[] before = connection.getCurrentPanTilt();
+            assertTrue(connection.relTilt(420));
+            Thread.sleep(2000);
+            int[] after = connection.getCurrentPanTilt();
+            assertWithinMaxOffset(before[0], after[0]);
+            assertWithinMaxOffset(before[1] + 420, after[1]);
         }
     }
 
     @Test
-    public void testRelFocus() {
+    public void testRelZoom() throws InterruptedException {
         if (doTests) {
-            fail("Not yet implemented");
+            int oldZoom = connection.getCurrentZoom();
+            assertTrue(connection.relZoom(65));
+            Thread.sleep(1000);
+            int newZoom = connection.getCurrentZoom();
+            assertWithinMaxOffset(oldZoom + 65, newZoom);
         }
     }
 
     @Test
-    public void testHasAutoFocus() {
+    public void testRelFocus() throws InterruptedException {
         if (doTests) {
-            fail("Not yet implemented");
+            assertTrue(connection.setAutoFocus(false));
+            int oldFocus = connection.getCurrentFocus();
+            assertTrue(connection.relFocus(65));
+            Thread.sleep(1000);
+            int newFocus = connection.getCurrentFocus();
+            assertWithinMaxOffset(oldFocus + 65, newFocus);
         }
     }
 
     @Test
-    public void testUpdate() {
+    public void testSetAutoFocus() {
         if (doTests) {
-            fail("Not yet implemented");
+            boolean currentAutoFocus = connection.hasAutoFocus();
+            connection.setAutoFocus(!currentAutoFocus);
+            assertEquals(!currentAutoFocus, connection.hasAutoFocus());
+        }
+    }
+
+    @Test
+    public void testUpdateAll() {
+        Camera c = new Camera();
+        connection = spy(new LiveCameraConnection("192.168.10.101"));
+        doReturn(new CameraSettings(0, 0, 0, 0)).when(connection).getCurrentCameraSettings();
+        doReturn(true).when(connection).absPanTilt(1965, 1965);
+        doReturn(true).when(connection).absZoom(1965);
+        doReturn(true).when(connection).absFocus(1965);
+        connection.update(c, new CameraSettings(1965, 1965, 1965, 1965));
```

Comment:
It might be better to use different values, otherwise when setting 2 is saved as setting 1 the test would still pass.